### PR TITLE
[twine-sugarcube] add extension methods for ReadonlyArray

### PR DESCRIPTION
**Pull request by ezsh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.motoslave.net/sugarcube/2/docs/#methods-array

### CODE REVIEW

File: types/twine-sugarcube/extensions.d.ts [hunk 1]
```diff
@@ -6,6 +6,111 @@ declare global {
...
+         * Returns the first member from the array. Does not modify the original.
+         * @since SugarCube 2.2.7.0
+         * @example
+         * // Given: $pies = ["Blueberry", "Cherry", "Cream", "Pecan", "Pumpkin"]
+         * $pies.first() // Returns "Blueberry"
+         */
+        first(): T | undefined;
+
+        /**
+         * Returns a new array consisting of the flattened source array (i.e. flat map reduce). Does not modify the original.
```
Comment (tmedwards) [inline types/twine-sugarcube/extensions.d.ts:48]: There should be a deprecation note/tag here.  E.g.,

```
* [DEPRECATED] Returns a new …
```

It doesn't have to be exactly that, though that's generally how I do it in the source code—e.g., [extensions.js#L1875](https://github.com/tmedwards/sugarcube-2/blob/27903a854b7bedbb9fa60efa33e47c99bab7a359/src/lib/extensions.js#L1875)—but I do want deprecated APIs marked as such for inline editor hints.

**Review by tmedwards — CHANGES_REQUESTED** [review]
See the line comment.

File: types/twine-sugarcube/extensions.d.ts (hunk 1 above)
Comment (ezsh) [reply types/twine-sugarcube/extensions.d.ts:48]: Done.

**Review by tmedwards — APPROVED** [review]

**Comment by ezsh** [comment]
Thank you, @tmedwards!

Ready to merge

**Comment by tmedwards** [comment]
Ready to merge

**Pull request merged**